manifests/init.pp

```diff
@@ -157,7 +157,7 @@
   # If undefined and selinux is present and not disabled, use selinux.
   case $mrepo::selinux {
     undef: {
-      case $selinux {
+      case fact('os.selinux.current_mode') {
```

Comment:
According to https://puppet.com/docs/puppet/latest/style_guide.html#referencing-facts this line should look like this:

```suggestion
      case $facts['os']['selinux']['current_mode'] {
```

Reply:
I'm using the stdlib [`fact`](https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/REFERENCE.md#factstring-fact_name) function here.  It has the advantage of returning `undef` instead of exploding if the eg. `os` or `os['selinux']` fact doesn't exist.  I *think* it's not strictly required in this case.  Even non selinux supporting OSes *should* have the base fact.  I probably need to bump the minimum required stdlib version though.  let me check...

Reply:
As explained in the stdlib commit that added the function.
> The fact() function allows dot-notation reference to facts. It is an
alternative to using $facts directly with array-indexing. Array-indexing
is often onerous to use since it doesn't align with how structured facts
are accessed elsewhere in the ecosystem and if any element in a
multi-step path doesn't exist, array indexing can cause a compilation
failure.

Reply:
Awesome! Somehow I was not aware of that function... I see much use of this in my future :)